dex/logging.go

```diff
@@ -20,6 +36,53 @@ type LoggerMaker struct {
 	Levels       map[string]slog.Level
 }
 
+// log contains the slog.Logger and fields needed to spawn subloggers. It
+// satisfies the Logger interface.
+type log struct {
+	slog.Logger
+	name    string
+	level   slog.Level
+	levels  map[string]slog.Level // may be nil
+	backend *slog.Backend
+}
+
+// SubLogger creates a new Logger for the subsystem with the given name. If name
+// exists in the levels map, use that level, otherwise the parent's log level is
+// used.
+func (lggr *log) SubLogger(name string) Logger {
+	combinedName := fmt.Sprintf("%s[%s]", lggr.name, name)
+	logger := lggr.backend.Logger(combinedName)
+	level := lggr.level
+	// If levels exists and name is in the map, use that level.
+	if lggr.levels != nil {
+		if lvl, ok := lggr.levels[name]; ok {
+			level = lvl
+		}
+	}
+	logger.SetLevel(level)
+	return &log{
+		Logger:  logger,
+		name:    combinedName,
+		level:   level,
+		levels:  lggr.levels,
+		backend: lggr.backend,
+	}
+}
+
+// StdOutLogger creates a Logger with the provided name with lvl as the log
+// level and prints to standard out.
+func StdOutLogger(name string, lvl slog.Level) Logger {
+	backend := slog.NewBackend(os.Stdout)
+	logger := backend.Logger(name)
+	logger.SetLevel(lvl)
+	return &log{
+		Logger:  logger,
+		name:    name,
+		level:   lvl,
+		backend: backend,
+	}
+}
+
 // NewLoggerMaker parses the debug level string into a new *LoggerMaker. The
 // debugLevel string can specify a single verbosity for the entire system:
 // "trace", "debug", "info", "warn", "error", "critical", "off".
```

Comment:
`NewLoggerMaker` could be improved further by accepting an `io.Writer` instead of a `slog.Backend`, and doing `slog.NewBackend(writer)` internally, which would allow most consumer packages to not require importing directly slog at all.